kem/kyber/kyber512/kyber.go

```diff
@@ -331,47 +331,47 @@ func (*scheme) DeriveKey(seed []byte) (kem.PublicKey, kem.PrivateKey) {
 	return NewKeyFromSeed(seed[:])
 }
 
-func (*scheme) Encapsulate(pk kem.PublicKey) (ct []byte, ss []byte) {
+func (*scheme) Encapsulate(pk kem.PublicKey) (ct, ss []byte, err error) {
 	ct = make([]byte, CiphertextSize)
 	ss = make([]byte, SharedKeySize)
 
 	pub, ok := pk.(*PublicKey)
 	if !ok {
-		panic(kem.ErrTypeMismatch)
+		return nil, nil, kem.ErrTypeMismatch
 	}
 	pub.EncapsulateTo(ct, ss, nil)
 	return
 }
 
 func (*scheme) EncapsulateDeterministically(pk kem.PublicKey, seed []byte) (
-	ct []byte, ss []byte) {
+	ct, ss []byte, err error) {
 	if len(seed) != EncapsulationSeedSize {
-		panic(kem.ErrSeedSize)
+		return nil, nil, kem.ErrSeedSize
 	}
 
 	ct = make([]byte, CiphertextSize)
 	ss = make([]byte, SharedKeySize)
 
 	pub, ok := pk.(*PublicKey)
 	if !ok {
-		panic(kem.ErrTypeMismatch)
+		return nil, nil, kem.ErrTypeMismatch
 	}
 	pub.EncapsulateTo(ct, ss, seed)
```

Comment:
EncapsulateTo (and other functions) also panics internally.
It seems that returning errors must go deeper, The package pke/kyber also panics.

Reply:
Those panics can't be triggered when called via the generic API.